src/main/java/baritone/cache/WorldProvider.java

```diff
@@ -75,7 +75,18 @@ public final void initWorld(ResourceKey<Level> worldKey, DimensionType world) {
             directory = new File(directory, "baritone");
             readme = directory;
         } else { // Otherwise, the server must be remote...
-            String folderName = mc.isConnectedToRealms() ? "realms" : mc.getCurrentServer().ip;
+            String folderName;
+            if (mc.isConnectedToRealms()) {
+                folderName = "realms";
+            } else {
+                if (mc.getCurrentServer() != null) {
+                    folderName = mc.getCurrentServer().ip;
+            } else {
+                //replaymod causes null currentServerData and false singleplayer.
+                currentWorld = null;
+                return;
+            }
```

Comment:
This was probably an issue that I should have noticed on the initial commit but the indentation here is really confusing, should you fix it in this pr, or should I open a new one to fix it?

Reply:
o oops. no that's an issue with indenting here specifically... I fix